lib/videojs-playlist-ui.js

```diff
@@ -1,25 +1,188 @@
+
+    // Thumbnail image
+    li.appendChild(createThumbnail(item.thumbnail));
+
+    // Duration
+    if (item.duration) {
+      let duration = document.createElement('time');
+      let time = videojs.formatTime(item.duration);
+      duration.className = 'vjs-playlist-duration';
+      duration.setAttribute('datetime', 'PT0H0M' + item.duration + 'S');
+      duration.appendChild(document.createTextNode(time));
+      li.appendChild(duration);
+    }
+
+    // Name and description
+    let name = document.createElement('cite');
+    name.className = 'vjs-playlist-name';
+    name.appendChild(document.createTextNode(item.name || this.localize('Untitled Video')));
+    li.appendChild(name);
+
+    if (item.description) {
+      let description = document.createElement('p');
+      description.className = 'vjs-playlist-description';
+      description.appendChild(document.createTextNode(item.description));
+      li.appendChild(description);
+    }
+    return li;
+  }
+});
+
+videojs.PlaylistMenu = videojs.Component.extend({
+  init: function(player, options) {
+
+    if (!player.playlist || player.playlist() === undefined) {
+      throw new Error('videojs-playlist is required for the playlist component');
+    }
+
+    if (!options.el) {
+      this.el_ = document.createElement('ol');
+      this.el_.className = options.className;
+      options.el = this.el_;
+    }
+
+    videojs.Component.call(this, player, options);
+    this.createPlaylist_();
+
+    if (!videojs.TOUCH_ENABLED) {
+      this.addClass('vjs-mouse');
+    }
+
+    player.on('loadstart', (event) => {
+      this.update();
+    });
+  },
+  createPlaylist_: function() {
+    const playlist = this.player_.playlist();
+
+    this.items = [];
+    for (let i in playlist) {
+      let item = new videojs.PlaylistMenuItem(this.player_, {
+        item: playlist[i]
+      });
+      this.items.push(item);
+      this.addChild(item);
+    }
+    // select the current playlist item
+    let selectedIndex = this.player_.playlist.currentItem();
+    if (selectedIndex >= 0) {
+      this.items[selectedIndex].el_.className += ' vjs-selected';
+    }
+  },
+  update: function() {
+    // reset the selection
+    const currentItem = this.player_.playlist.currentItem();
+    for (let i = 0; i < this.items.length; i++) {
+      if (i === currentItem) {
+        this.items[i].addClass('vjs-selected');
+      } else {
+        this.items[i].removeClass('vjs-selected');
+      }
+    }
+  }
+});
+
+/**
+ * Initialize the plugin.
+ * @param options (optional) {object} configuration for the plugin
+ */
+const playlistUi = function(options) {
+  const player = this;
+  let settings, elem;
+
+  if (!player.playlist || player.playlist() === undefined) {
+    throw new Error('videojs-playlist is required for the playlist component');
+  }
+
+  // if the first argument is a DOM element, use it to build the component
+  if ((typeof HTMLElement !== 'undefined' && options instanceof HTMLElement) ||
```

Comment:
I think it'll be better to always require an options object for this.
Then, either there's an `options && options.el` or there isn't.